### PR TITLE
chore: enable usage of abstr elements as popup openers

### PR DESCRIPTION
**Pull request by ilhan007**
There is a use-case brought with this issue https://github.com/SAP/ui5-webcomponents/issues/10126 to open a menu (popover-based component) by an abstract UI5Element used as an opener (ui5-toolbar-item) as shown in the example:

```html
<ui5-toolbar>
	<ui5-toolbar-button icon="employee" text="Left 3"></ui5-toolbar-button>
	<ui5-toolbar-separator></ui5-toolbar-separator>
	<ui5-toolbar-button icon="decline" text="Left 4"></ui5-toolbar-button>
	<ui5-toolbar-button id="btnOpenMenu" text="Open Menu" prevent-overflow-closing></ui5-toolbar-button>
</ui5-toolbar>
		
<ui5-menu header-text="Basic Menu with Items" id="menu" opener="btnOpenMenu">
	<ui5-menu-item text="New File" icon="add-document"></ui5-menu-item>
	<ui5-menu-item text="New Folder" icon="add-folder" disabled></ui5-menu-item>
	<ui5-menu-separator></ui5-menu-separator>
	<ui5-menu-item text="Open" icon="open-folder"></ui5-menu-item>
	<ui5-menu-item text="Close"></ui5-menu-item>
</ui5-menu>
```
```js
const btnOpenMenu = document.getElementById("btnOpenMenu");
const menu = document.getElementById("menu");

btnOpenMenu.addEventListener("ui5-click", function(event) {
	menu.open = !menu.open;
});
```

The change tries to use getDomRef if exists to get to the real DOM reference - the root for ordinary ui5 web comps or  the actual rendered element for an abstract ui5 web comps.

```ts
return (openerHTMLElement as UI5Element)?.getDomRef?.() || openerHTMLElement;
```

Fixes: https://github.com/SAP/ui5-webcomponents/issues/10126

### CODE REVIEW

File: packages/main/src/Popover.ts [hunk 1]
```diff
@@ -303,11 +303,9 @@ class Popover extends Popup {
 		}
 
 		const rootNode = this.getRootNode();
+		const openerHTMLElement = rootNode instanceof Document ? rootNode.getElementById(opener) : document.getElementById(opener);
 
-		if (rootNode instanceof Document) {
-			return rootNode.getElementById(opener);
-		}
-		return document.getElementById(opener);
+		return (openerHTMLElement as UI5Element)?.getFocusDomRef?.() || openerHTMLElement;
```
Comment (dimovpetar) [inline packages/main/src/Popover.ts:308]: Please make it explicit by using `instanceOfUI5Element`

File: packages/main/src/Popover.ts
Comment (dimovpetar) [inline packages/main/src/Popover.ts:1]: Please add test, ideally dedicated one in Popover.spec

**Review by dimovpetar — DISMISSED** [review]
I like the idea to use `getFocusDomRef`. It seems suitable for all abstract elements that I know of, except for the `ui5-tab`. There, `getDomRefInStrip` should be used. However, this is not a valid usage, so just ignore this comment.

**Comment by dobrinyonkov** [comment]
Toolbar related changes look good to me.

**Review by TeodorTaushanov — APPROVED** [review]

**Review by nnaydenow — APPROVED** [review]
LGTM. Lets discuss how to remove renderFinished call and address it in separate PR.

**Review by dimovpetar — APPROVED** [review]

**Review by nnaydenow — APPROVED** [review]

**Pull request merged**